src/include/planner/plan_util.h

```diff
@@ -149,5 +169,92 @@ inline void PlanUtil::GetTablesReferenced(const planner::AbstractPlan *plan,
   }
 }
 
+namespace {
+
+template <typename T>
+bool isDisjoint(const std::set<T> &setA, const std::set<T> &setB) {
+  bool disjoint = true;
+  if (setA.empty() || setB.empty())
+    return disjoint;
+
+  typename std::set<T>::const_iterator setA_it = setA.begin();
+  typename std::set<T>::const_iterator setB_it = setB.begin();
+  while (setA_it != setA.end() && setB_it != setB.end() && disjoint) {
+    if (*setA_it == *setB_it)
+      disjoint = false;
+    else if (*setA_it < *setB_it)
```

Comment:
It seems that you're assuming an ascending order for both sets, which may not be true. I think you need to sort the two sets before doing this.

Reply:
Hi,

The std::set stores the element inside in an ascending order already by default; so, there's no need to sort explicitly here.

Reply:
But actually, yeah. Let's sort them to be safe. Though the current implementation provide only sorted sets to this utility function, being a utility function, our users can give us all sorts of crazy input sets; so, better safe than sorry :)